woocommerce/wc-admin
# Create task list APIs

Sorry for the size of this PR.  The majority of the LoC is simply extracting code into new classes, but there are some notable changes:

* TaskLists/TaskList API
* Migrating hidden lists in a single option
* Adding backwards compatibility for getter/setters for old hidden options

Follow-up PRs:

* Updating frontend to use the new API instead of getting options directly
* Extraction of individual tasks into separate classes
* Task API
* Onboarding data store actions

### Detailed test instructions:

1. Bump your WooCommerce Admin version to trigger an install/update
4. Make sure any old versions of `woocommerce_task_list_hidden` or `woocommerce_extended_task_list_hidden` have been appropriately updated into the new option `woocommerce_task_list_hidden_lists` as an array of IDs of the task lists (currently this should only be "setup" for the main task list if it's hidden).
5. Attempt to update `woocommerce_task_list_hidden` with the value `yes` or `no`
6. Make sure the updates appear as an array of task list IDs in `woocommerce_task_list_hidden_lists`
7. Disable the `tasks` feature flag and smoke test the task list and onboarding pages
